src/test/java/io/personium/test/jersey/cell/auth/AuthErrorTest.java

```diff
@@ -60,6 +60,27 @@ public AuthErrorTest() {
         super(new PersoniumCoreApplication());
     }
 
+    /**
+     * Testing for returning 400 when unsupported grant_type is passed
+     */
+    @Test
+    public final void Returns400_WhenUnsupportedGrantTypesIsPassed() {
+        String unsupported_grant_type = "unsupported_grant_type";
+
+        TResponse passRes = Http.request("authn/auth.txt")
+                .with("remoteCell", TEST_CELL1)
+                .with("body", "grant_type=" + unsupported_grant_type)
+                .returns()
+                .statusCode(HttpStatus.SC_BAD_REQUEST);
+
+        AuthTestCommon.checkAuthenticateHeaderNotExists(passRes);
+        String code = PersoniumCoreAuthnException.CLIENT_ASSERTION_PARSE_ERROR.getCode();
+        String message = PersoniumCoreAuthnException.CLIENT_ASSERTION_PARSE_ERROR.getMessage();
+        String errDesc = String.format("[%s] - %s", code, message);
+
+        checkErrorResponseBody(passRes, String.format(Error.UNSUPPORTED_GRANT_TYPE, unsupported_grant_type), errDesc);
```

Comment:
Where does it check the message `Unsupported grant type. [unsupported_grant_type]`?